_sources/Tuples/TupleAssignmentwithunpacking.rst

```diff
@@ -139,6 +139,36 @@ More generally, if you have a list of tuples that each has more than two items,
 loop pulling out information from the tuples, the code will be far more readable if you unpack them into separate
 variable names right after the word ``for``.
 
+
+The Pythonic Way to Enumerate Items in a Sequence
```

Comment:
I'm not sure what to do here.  We had reached a compromise on this before because I was using enumerate in some labs that were earlier in the class and  so this material was moved up.  Now you are undoing that and students will not have seen this for the project.

I understand the argument about magic and I like to minimize the magic too, but this one seems really quite harmless.

Reply:
How about if you make a new page for enumerate and include it in this chapter if you want it to be available for labs at this point?

At this point in your courses, do your labs require students to be able to read and write the construction: `for idx, item in enumerate(seq):` at this point in the course? We explicitly teach them about tuple unpacking a few chapters on, and you'll see I added a part of a page explaining that construction as part of this PR.

Reply:
Its an interesting problem.  We have to order the chapters and some later things are definitely harder than earlier things.  But some are not, they are just more esoteric maybe... Not sure.  but, many of the courses using FOPP in the Applied Computing program don't get past chapter 12 in one semester.

What I have done as a compromise for some of the projects is to make the first part of the functions chapter an option to read much earlier, for those that really want to teach functions earlier.

I could do the same thing here and just point to this tuple unpacking chapter.  The labs can be done without enumerate, its just a nice construct that actually makes the code cleaner and easier for the students to understand and seems to lead to less errors in their code than having to iterate over only the index.